src/repr/src/scalar.rs

```diff
@@ -54,7 +55,14 @@ pub enum Datum<'a> {
     Bytes(&'a [u8]),
     /// A sequence of Unicode codepoints encoded as UTF-8.
     String(&'a str),
+    /// A variable-length multidimensional array of datums.
+    ///
+    /// Unlike [`Datum::List`], arrays are like matrices and are not permitted
```

Comment:
Typically I have seen n-dim arrays be called "tensors", not "matrices".

Reply:
Fixed!